ls1intum/Artemis
# `Development`: Unify public endpoints and introduce basic authorization test

## Release Notes
Changes to the API:

| Affected Methods | Previous Endpoint                  | New Endpoint                                   | Related External Functionality              |
| -------------------- | ------------------------------- | --------------------------------------- | ------------------------------------------ |
| PUT                        | api/admin/management/feature-toggle | api/admin/feature-toggle | Probably nothing                              |
| GET                        | time                                       | api/public/time                                  |  Probably nothing                               |
| POST                     | api/register                            | api/public/register                             | Probably nothing                                 |
| GET                       | api/activate                            | api/public/activate                             | Probably nothing                                 |
| POST                     | api/account/reset-password/init | api/public/account/reset-password/init | Probably nothing                      |
| POST                     | api/account/reset-password/finish | api/public/account/reset-password/finish | Probably nothing               |
| POST                     | api/logout                              | api/public/logout                              | Probably nothing                                  |
| GET                        | api/system-notifications/active | api/public/system-notifications/active | Probably nothing                          |
| POST                     | api/account/change-language | api/public/account/change-language | Probably nothing                            |
| GET, PUT                | managment/logs                   | api/admin/logs                                 | External Scripts/Documentation          |
|       POST               | api/saml2                                | api/public/saml2                               | External Applications                           |
| GET                       | api/account                            | api/public/account                             | External Applications ~~Orion~~       |
| GET, POST             | api/authenticate                    | api/public/authenticate                     | External Applications ~~Orion~~        |
| POST                      | api/athene-result/:exerciseId | api/public/athene-result/:exerciseId | Athene                                                 |
| POST                      | api/lti/launch/:exerciseId       | api/public/lti/launch/:exercseId        | Platforms connected to Artemis via LTI|
| POST                      | api/lti13/auth-callback          | api/public/lti13/auth-callback           | Platforms connected to Artemis via LTI |
| POST                      | api/programming-submissions/:participationId | api/public/programming-submissions/:participationId | VCS (Bitbucket, Gitlab) |
| POST                      | api/programming-exercises/test-cases-changed/:exerciseId | api/public/programming-exercises/test-cases-changed/:exerciseId | VCS (Bitbucket, Gitlab) |
| POST                     | api/programming-exercises/new-result | api/public/programming-exercises/new-result | CI (Bamboo, Jenkins) |
| POST                     | api/programming-exercises/:exerciseId/build-plan | api/public/programming-exercises/:exerciseId/build-plan | CI (Gitlab CI) |





## Checklist
#### General


#### Server

### Motivation and Context


This follows up #5574 and #5638 and covers all public endpoints. I also add a first authorization test to check the correct annotation usage. This setup can be extended in the future with more extensive tests.

### Description

I implemented a first authorization test to check the correct usage of the pre-authorization annotations. It runs for all profile setups separately, however, subsequent setups only test relevant endpoints. This way, all endpoints only get tested once. I included a new annotation for public endpoints and marked all not annotated endpoints (flagged by the test) after validating the flag, of course.

### Steps for Testing


- Access and log into Artemis.
This should validate that the annotation works. Please look at the changed occurrences and test more in case you feel it's necessary.

- Test commit and build for programming exercises
This makes sure that the legacy resource is still working

### Steps for Code Review

Please check all replacements for consistency. The most relevant part is then the test and test service. If you have any suggestions on how to improve the structure, especially to be more extensible, feel free.

### Review Progress





#### Code Review
#### Manual Tests
